### PR TITLE
[Merged by Bors] - feat: first definitions in the API for the homology of short complexes

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, some basic constructions for left and right homology of short complexes are combined.

---
- [x] depends on: #6230
- [x] depends on: #6231

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/ShortComplex/Homology.lean [hunk 1]
```diff
@@ -33,22 +33,171 @@ namespace CategoryTheory
...
   iso : left.H ≅ right.H
+  /-- the pentagon relation expressing the compatibility of the left
+  and right homology data -/
   comm : left.π ≫ iso.hom ≫ right.ι = left.i ≫ right.p := by aesop_cat
 
 attribute [reassoc (attr := simp)] HomologyData.comm
 
+variable (φ : S₁ ⟶ S₂) (h₁ : S₁.HomologyData) (h₂ : S₂.HomologyData)
+
+/-- An homology map data for a morphism `φ : S₁ ⟶ S₂` where both `S₁` and `S₂` are
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/Homology.lean:59]:
```suggestion
/-- A homology map data for a morphism `φ : S₁ ⟶ S₂` where both `S₁` and `S₂` are
```

File: Mathlib/Algebra/Homology/ShortComplex/Homology.lean [hunk 2]
```diff
@@ -33,22 +33,171 @@ namespace CategoryTheory
...
+equipped with homology data consist of left and right homology map data. -/
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/Homology.lean:60]:
```suggestion
equipped with homology data consists of left and right homology map data. -/
```

File: Mathlib/Algebra/Homology/ShortComplex/Homology.lean [hunk 3]
```diff
@@ -33,22 +33,171 @@ namespace CategoryTheory
...
+    S.HomologyData where
+  left := LeftHomologyData.ofZeros S hf hg
+  right := RightHomologyData.ofZeros S hf hg
+  iso := Iso.refl _
+
+/-- If `φ : S₁ ⟶ S₂` is a morphism of short complexes such that `φ.τ₁` is epi, `φ.τ₂` is an iso
+and `φ.τ₃` is mono, then a homology data for `S₁` induces a homology data for `S₂`.
+The inverse construction is `ofEpiOfIsIsoOfMono'`. -/
+@[simps]
+noncomputable def ofEpiOfIsIsoOfMono (φ : S₁ ⟶ S₂) (h : HomologyData S₁)
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/Homology.lean:153]: Not sure if this is useful, but `ofEpiOfIsIsoOfMono` and `ofEpiOfIsIsoOfMono'` could be bundled together into an equiv.

Comment (joelriou) [reply Mathlib/Algebra/Homology/ShortComplex/Homology.lean:155]: Indeed, I do not see situations where an equality between two `HomologyData` would be useful. Relations between `Left/Right/HomologyData` are better expressed using `HomologyMapData`. (Actually, in early versions of this code, `HomologyMapData` were morphisms in the category of short complexes equipped with a `HomologyData`.)

File: Mathlib/Algebra/Homology/ShortComplex/Homology.lean [hunk 4]
```diff
@@ -33,22 +33,171 @@ namespace CategoryTheory
 
 open Category Limits
 
-variable {C : Type*} [Category C] [HasZeroMorphisms C] (S : ShortComplex C)
+variable {C : Type _} [Category C] [HasZeroMorphisms C] (S : ShortComplex C)
```
[the remark this answers is not in the document]
Comment (joelriou) [reply Mathlib/Algebra/Homology/ShortComplex/Homology.lean]: This code was written before this policy change, and I am late jumping on the train...

Thanks for the review! When CI is done, it should be ok.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**